app/models/user.rb

```diff
@@ -1,2 +1,47 @@
 class User < ActiveRecord::Base
+  has_many :categories
+  has_many :creators, through: :categories
+  has_many :tweets, through: :creators
+  has_many :videos, through: :creators
+  validates_presence_of :name, :uid, :provider
+  validates_uniqueness_of :uid
+
+ def self.find_or_create_from_omniauth(auth_hash)
+   user = self.find_by(uid: auth_hash["uid"], provider: auth_hash["provider"])
+   if !user.nil?
+     # User found continue on with your life
```

Comment:
Same as the comment above regarding `!nil?`
Also, nice use of comments to let others know what your code is doing :+1: